primitives/state-machine/src/lib.rs

```diff
@@ -1005,7 +994,8 @@ mod tests {
 		overlay.commit_prospective();
 
 		assert_eq!(
-			overlay.committed,
+			overlay.changes(None).map(|(k, v)| (k.clone(), v.value().cloned()))
+				.collect::<HashMap<_, _>>(),
```

Comment:
This is not strictly testing the same thing, the previous test was asserting that the effect of `commit_prospective`.
An easy way to do it would be to call `overlay.set_pending(Default::default)` before the assertion.

Reply:
Isn't the `commit_prospective` above the assert achieving the same effect?

Reply:
commit prospective is merging both layer, the test assert that all the resulting values are in the bottom layer.
So your new version does not test this anymore, dropping the prospective layer will again test this.

Reply:
The commit was already done in the original test. Just dropping it here would brake the test.

Reply:
I did not mean drop the commit, but drop the prospective.

But yes that is probably not the original intention of the test (thus the 'strictly' in my original comment), so it does not matter that much :)